main.cpp

```diff
@@ -169,7 +176,39 @@ static int parse_argv(int argc, char *argv[], const char **uart)
         return -EINVAL;
     }
 
-    *uart = argv[optind++];
+    if (stat(argv[optind], &st) == -1) {
```

Comment:
Fair... but isn't this kind of detailed checking asking for a TOCTOU report from static analysis?